src/program/snabblwaftr/lib/util.lua

```diff
@@ -0,0 +1,23 @@
+module(..., package.seeall)
+
+function file_exists(filename)
+   local f = io.open(filename, "r")
+   if f then
+      f:close()
+      return true
+   end
+   return false
+end
+
+function dir_exists(dirname)
```

Comment:
Wow, this is a bit obscure (in particular the `code == 21` with the magic number there). We can do something like this instead:
```lua
local syscall = require("syscall")

function file_exists(path)
   local stat = syscall.stat(path)
   return stat and stat.isreg
end

function dir_exists(path)
   local stat = syscall.stat(path)
   return stat and stat.isdir
end
```

It is okay to keep it that way, but I am not completely sure if it's worth keeping a separate module for such small functions maybe the call sites can use `syscall.stat()` directly. Just some food for thought.